### DISCUSSION TITLE
Is it possible to control a 3 phase motor with a cheap 8 bit MCU?

### DISCUSSION THREAD

**Discussion by LaZsolt** [dbuezas/lgt8fx#246] [Ideas]
I was thinking a bit about, is it possible to control a 3 phase motor with LGT8F328P?
The question is only theoretical, I don't want to implement it.

In this MCU there are 4 timers. One for controllig the rotation speed at least at start up. Other three for generating outputs for MOSFETs.
For controllig the rotation speed Timer 2 could provide the clock signal for the other three timers, because Timer 2 has no external clock input and no dead time capability. IN CTC mode Timer 2 output "A" could be the source of other three timers external clock. It must be wired between pins. The three timers "A" and "B" outputs could provide the MOSFETs controlling signals with dead time.

But there is still a problem. When the motor starting at low frequency the MOSFETS is open too long, there a big current flow will develop. So there must be a three phase PWM also.

Is it possibile?

**Comment by ithinkido** [comment]
BLDC motos use three phase power and some use an 8 bit MCU for control This is an example project, 

[SimpleFOC](https://www.simplefoc.com/)

**Comment by dbuezas** [comment]
So you want to do everything purely with hw timers? That's an interesting challenge. Three timers with dead time sound like a good fit for generating the phases, but I don't see either how to add pwm to the mix (except in software ofc). 
Some drivers (the drv8833 for example) have inbuilt current limiting via an internally triggered pwm, that may do the trick. I think It also adds dead time by itself and is by design incapable of shorting both high and low side mosfets at the same time, so it kind of defeats 
the purpose of the challenge.

**Comment by hmeijdam** [comment]
Not sure if I understand your question, but many Electronic Speed Controllers (ESC) for RC planes, helicopters or drones used Atmega8. 
[Here](https://docs.google.com/spreadsheets/d/13tMlu5ldLNpZXwbe6UhDHJhcgTVuljm8HDiDp9WO9Pk/edit#gid=0) is a table which ones
and [here](https://github.com/bitdump/BLHeli/tree/master/Atmel) ESC firmware for Atmega8

**Comment by LaZsolt** [comment]
Thank you the valuable comments. I didn't know about these solutions. 
I thought the LGT would drive the MOSFETs directly, but a motor driver much cheaper and time and PCB space saving solution.

**Reply by dbuezas** [reply]
Because of the high current outputs you mean, right? They work very well driving mosfets directly

**Reply by LaZsolt** [reply]
yes

**Comment by jokokiswan** [comment]
Hi guys did you have a sample doing deadtime pwm on lgt8f328? I did duo pwm sample on lgt8f328 got error message no pwm duo declaration or something using this package on arduino IDE. Thank you.

**Reply by LaZsolt** [reply]
Try this code: https://dbuezas.github.io/arduino-web-timers/#mcu=LGT8F328P&timer=1&DeadTime=on&OCR1B=128&DTR1L=16&DTR1H=19&CompareOutputModeB=clear-on-match%2C+set-at-max&CompareOutputModeA=clear-on-match%2C+set-at-max&clockPrescalerOrSource=8&FCPU_UI=16Mhz

**Reply by dbuezas** [reply]
Heads up, the cpu clock is expected at 16MHz in the link

**Reply by jokokiswan** [reply]
Thank you bro!, Got many plus2 than original 328, very good for bare metalling things.
